packages/react-components/react-toast/src/components/Toast.tsx

```diff
@@ -99,7 +99,7 @@ export const Toast: React.FC<ToastProps & { visible: boolean }> = props => {
     <Transition in={visible} unmountOnExit mountOnEnter timeout={500} onExited={remove} nodeRef={toastRef}>
       <div ref={toastRef} className={mergeClasses(styles.toast, visible && styles.slide, !visible && styles.fadeOut)}>
         {children}
-        <Timer onTimeout={close} timeout={timeout ?? -1} running={running} />
+        <Timer key={updateId} onTimeout={close} timeout={timeout ?? -1} running={running} />
```

Comment:
_**any**_ update will reset the timer. It would increase the complexity to make sure the timer did not reset, but I don't think it's worth it until someone reports it